### ISSUE TITLE
CoderException: cannot encode a null String with Beam 2.36+

### ISSUE THREAD

**Issue by turb** [spotify/scio#4284]
One of our jobs stopped working on Dataflow after an upgrade to scio 0.11.5.

After some checks, it is not caused by scio itself: we have the same error on scio 0.11.4 when upgrading only Beam to 2.36 (and same with 2.37). scio 0.11.4 + Beam 2.35 works fine.

I was aiming to open a bug on the Beam side, however I choose to open one here first, in case it rings a bell.

Stacktrace:
```
Caused by: org.apache.beam.sdk.coders.CoderException: cannot encode a null String
	at Due to Exception while trying to `encode` an instance of Tuple2: Can't decode field _2.(:0)
	at org.apache.beam.sdk.coders.StringUtf8Coder.encode(StringUtf8Coder.java:74)
	at org.apache.beam.sdk.coders.StringUtf8Coder.encode(StringUtf8Coder.java:68)
	at org.apache.beam.sdk.coders.StringUtf8Coder.encode(StringUtf8Coder.java:37)
	at com.spotify.scio.coders.instances.Tuple2Coder$$anonfun$encode$4.apply$mcV$sp(TupleCoders.scala:57)
	at com.spotify.scio.coders.instances.Tuple2Coder$$anonfun$encode$4.apply(TupleCoders.scala:57)
	at com.spotify.scio.coders.instances.Tuple2Coder$$anonfun$encode$4.apply(TupleCoders.scala:57)
	at com.spotify.scio.coders.instances.Tuple2Coder.onErrorMsg(TupleCoders.scala:41)
	at com.spotify.scio.coders.instances.Tuple2Coder.encode(TupleCoders.scala:57)
	at com.spotify.scio.coders.instances.Tuple2Coder.encode(TupleCoders.scala:35)
	at com.spotify.scio.coders.WrappedBCoder$$anonfun$encode$1.apply$mcV$sp(Coder.scala:269)
	at com.spotify.scio.coders.WrappedBCoder$$anonfun$encode$1.apply(Coder.scala:269)
	at com.spotify.scio.coders.WrappedBCoder$$anonfun$encode$1.apply(Coder.scala:269)
	at com.spotify.scio.coders.WrappedBCoder.catching(Coder.scala:259)
	at com.spotify.scio.coders.WrappedBCoder.encode(Coder.scala:269)
	at org.apache.beam.sdk.coders.Coder.encode(Coder.java:136)
	at org.apache.beam.sdk.transforms.join.UnionCoder.encode(UnionCoder.java:74)
	at org.apache.beam.sdk.transforms.join.UnionCoder.encode(UnionCoder.java:32)
	at org.apache.beam.sdk.util.WindowedValue$FullWindowedValueCoder.encode(WindowedValue.java:607)
	at org.apache.beam.sdk.util.WindowedValue$FullWindowedValueCoder.encode(WindowedValue.java:558)
	at org.apache.beam.runners.dataflow.worker.ShuffleSink$ShuffleSinkWriter.encodeToChunk(ShuffleSink.java:331)
<snip>
	at ### Coder materialization stack ###.(:0)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:66)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:67)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:66)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:67)
	at com.spotify.scio.coders.CoderMaterializer$.beam(CoderMaterializer.scala:49)
	at com.spotify.scio.coders.CoderMaterializer$.beam(CoderMaterializer.scala:35)
	at com.spotify.scio.values.SCollection.parDo(SCollection.scala:230)
	at com.spotify.scio.values.SCollection.parDo$(SCollection.scala:227)
	at com.spotify.scio.values.SCollectionImpl.parDo(SCollection.scala:1603)
	at com.spotify.scio.values.SCollection.map(SCollection.scala:712)
	at ### Coder materialization stack ###.(:0)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:61)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:67)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:67)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:66)
	at com.spotify.scio.coders.CoderMaterializer$.beamImpl(CoderMaterializer.scala:67)
	at com.spotify.scio.coders.CoderMaterializer$.beam(CoderMaterializer.scala:49)
	at com.spotify.scio.coders.CoderMaterializer$.beam(CoderMaterializer.scala:35)
	at com.spotify.scio.values.SCollection.parDo(SCollection.scala:230)
	at com.spotify.scio.values.SCollection.parDo$(SCollection.scala:227)
	at com.spotify.scio.values.SCollectionImpl.parDo(SCollection.scala:1603)
```

**Comment by turb** [comment]
I tried a bunch of times to isolate the problem, without success.

However, the error does not happen anymore with Beam 2.38.

I suspect it had to do with something linked to https://github.com/apache/beam/pull/17147 however this one is not included in 2.38 release.

**Comment by brucearctor** [comment]
@turb -- do you think we can close this as an issue?  Sounds like not a problem that Scio needs to address, since resolved with the recent Beam release?

**Comment by turb** [comment]
@brucearctor yes, I did not close it previously to see if someone had any insight in this.

**Comment by clairemcginty** [comment]
Hi! yes there is a known [CoGroup regression](https://issues.apache.org/jira/browse/BEAM-14171) with the version of Beam in Scio 0.11.5. Scio 0.11.6 has just been released and includes Beam 2.38.0, which has a fix for this issue.
